detectron2/utils/events.py

```diff
@@ -295,6 +302,42 @@ def put_scalars(self, *, smoothing_hint=True, **kwargs):
         for k, v in kwargs.items():
             self.put_scalar(k, v, smoothing_hint=smoothing_hint)
 
+    def put_histogram(self, hist_name, hist_tensor, bins=1000):
+        """
+        Create an histogram of a numpy array, calculate its parameters and add them
+        to self `_histograms`.
+
+        Args:
+            hist_name (str): The name of the histogram to put into tensorboard.
+            hist_tensor (numpy.array): An `uint8` or `float` array
```

Comment:
Is this a tensor or an array?

Reply:
This is a NumPy array of the shape of the given parameter/gradient.

Possible usage could be:
```
storage.put_histogram(hist_name, value.data.cpu().numpy())
```
where value is a PyTorch tensor.

Reply:
Maybe if you implemented this with `torch.histc` the whole thing could be tensors?

Reply:
Good point! I will look into it :)

Reply:
I have now implemented the same functionality using only PyTorch functions. For calculating ```hist_edges``` I followed the implementation of ```np.histogram()``` to make sure the behavior is the same.